presto-tests/src/test/java/com/facebook/presto/tests/TestMinWorkerRequirement.java

```diff
@@ -0,0 +1,31 @@
+package com.facebook.presto.tests;
+
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunner;
+import static org.testng.Assert.fail;
+
+public class TestMinWorkerRequirement
+{
+
+    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "Insufficient active worker nodes")
+    public void testInsufficientWorkerNodes()
+            throws Exception
+    {
+        // TpchQueryRunner have 4 worker nodes by default
+        try (DistributedQueryRunner queryRunner = createQueryRunner(ImmutableMap.of(), ImmutableMap.of("query.min-worker-nodes", "5"))) {
+            queryRunner.execute("SELECT 1");
+            fail("Expect to throw a PrestoException");
+        }
+    }
+
+    @Test
+    public void testSufficientWorkerNodes()
```

Comment:
Can we add a test with first having enough nodes and then some nodes crashed so that we are below the limit but can still run queries? One possible solution could be binding a mock `InternalNodeManager` with getNodes overridden by your own method. You are welcome to come up with easier solution.